torch/_refs/__init__.py

```diff
@@ -620,11 +622,10 @@ def log10(a):
 def log_softmax(
     a: TensorLikeType,
     dim: int,
-    *,
```

Comment:
How did you discover this?

The dtype can actually be passed as a positional argument:
https://github.com/pytorch/pytorch/blob/d2b8b8f29121ed23e2b39446d09bba3a7eb96684/aten/src/ATen/native/native_functions.yaml#L3116

But `PythonRefInfo("_refs.log_softmax", ...)` picks up the OpInfo variant without positional dtype and this becomes untested. There two OpInfo for log_softmax:
https://github.com/pytorch/pytorch/blob/d2b8b8f29121ed23e2b39446d09bba3a7eb96684/torch/testing/_internal/common_methods_invocations.py#L14703-L14719

PythonRefInfo("_refs.log_softmax", ...) should use the variant with `partial(sample_inputs_softmax_variant, with_dtype=True)`

Reply:
> How did you discover this?

maybe i'm misunderstanding the comment in the yaml file, then (i quoted it as well in the commit message). my reading of it was "do not use the star here before dtype unlike in other funcs", which is why i removed it from the refs (ditto for softmax). is this not what it says?

Reply:
And the next step would be to make sure it's properly tested.